lib/astarte_vmq_plugin.ex

```diff
@@ -272,4 +288,60 @@ defmodule Astarte.VMQ.Plugin do
       {:error, {:already_started, pid}} -> pid
     end
   end
+
+  defp maybe_authorize_registration(mountpoint, username) do
+    [realm, device_id] = String.split(username, "/")
+    {:ok, decoded_device_id} = Device.decode_device_id(device_id, allow_extended_id: true)
+
+    case {exists?(realm, decoded_device_id), is_device_being_deleted?(realm, decoded_device_id)} do
+      {true, false} ->
+        {:ok, registration_modifiers(mountpoint, username)}
+
+      {true, true} ->
+        {:error, :device_deletion_in_progress}
+
+      {false, _} ->
+        {:error, :device_does_not_exist}
+    end
+  end
+
+  defp registration_modifiers(mountpoint, username) do
```

Comment:
You removed the `TODO` about these, and they still have to be evaluated thoroughly. Either restore it or open an issue to track this.